### ISSUE TITLE
Cannot open changed code in the source control panel

### ISSUE THREAD

**Issue by brucejo75** [Microsoft/vscode#43428]
Cannot open source code changes in the **Source Control** panel.
[Weird nobody has reported this yet]

### Environment
- VSCode Version: `1.20.0`, Version `1.21.0-insiders edition`
- OS Version: Windows 10 Version `1709 (OS Build 16299.214)`
- Git Version: WSL: `2.7.4`, Windows: `2.15.0.windows.1`

### Reproduction Steps

1.  Installed v1.20.0
2.  Go to a pre-existing project.
3.  Modify some files in my project
4.  Go to the **Source Control** panel.
5.  Click on a file in the panel.  Nothing happens.
6.  Right click, select `Open Changes`, Nothing happens.

**Note 1**: Right click, `Open File` works fine.

**Note 2**: I tried creating a brand new project, then doing a `git init`.  Made some file changes and the issue does not reproduce. 

**Note 3**:  This issue occurs in both my WSL based projects and Windows based projects.

Does this issue occur when all extensions are disabled?: **Yes**

**Comment by brucejo75** [comment]
I was playing around with this and noticed that `Open File (Head)` was failing for many of my changed files.  For files where `Open File (Head)` was working, `Open Changes` works fine.

**Comment by joaomoreno** [comment]
Can you open the developer tools (<kbd>F1</kbd>, `Developer Tools`) and see if something comes up when you click on a file in the panel?

Note that WSL is not currently supported. Its success in operating with VS Code is purely coincidental. Please use Git for Windows for testing this feature.

**Comment by brucejo75** [comment]
@joaomoreno, thanks for the response.

For my erroring repo I had used windows git not WSL git.  I think that is why it works on 1.19.3 and continues to work on v1.19.3.

It is true that I am using WSL, but I do not think I had used any WSL git commands on the failing project.  I pretty much use VS code and Source Tree (both on Windows).

Here is what I see with Developer tools:

```console
console.ts:123 [Extension Host] oh no (at Console.originalConsole.(anonymous function) [as error]
(c:\Program Files\Microsoft VS Code Insiders\resources\app\extensions\git\node_modules\diagnostic-channel-publishers\dist\src\console.pub.js:42:39))

  | t.log | @ | console.ts:123
-- | -- | -- | --
  | t._logExtensionHostMessage | @ | extensionHost.ts:394
  | (anonymous) | @ | extensionHost.ts:211
  | emitTwo | @ | events.js:106
  | emit | @ | events.js:194
  | process.nextTick | @ | internal/child_process.js:766
  | _combinedTickCallback | @ | internal/process/next_tick.js:73
  | _tickCallback | @ | internal/process/next_tick.js:104


```

**Comment by joaomoreno** [comment]
> Note 2: I tried creating a brand new project, then doing a git init. Made some file changes and the issue does not reproduce.

This tells me it's somewhat related to your workspace.

Is your workspace in some network share? Is there anything special with the path of that workspace?

**Comment by brucejo75** [comment]
No, all projects are on the same local SSD drive (E:).  And to be clear.  1.19.3 works on all of these projects without issue.

My project does have 4 sub-modules and I notice the new UI supports browsing the 4 sub-module git projects.

**Comment by joaomoreno** [comment]
But just to confirm, it's just that one project which doesn't work, correct?

**Comment by brucejo75** [comment]
So far I have only tested it on this project.  Multiple cloned instances, but correct just this one project.

Maybe I need to work on a repro...

**Comment by joaomoreno** [comment]
Yeah, we gotta figure out what is so special about that project. Let me know if you come up with some more details.

**Comment by joaomoreno** [comment]
@brucejo75 says:

> @joaomoreno,
> 
> I found and fixed my issue in #43428.  I thought I would write up my fix and how I debugged it hoping to help someone else out in this situation. [I could not add any comments to #43428] because it is closed.
> 
> version `1.19.3` worked for me but all versions after that failed.  version `1.20` was where submodule support was introduced into **vscode**.  Which made submodules suspicious.
> 
> ### Essentially my `.gitmodules` file was messed up.
> 1. I had previously tried to create a number of sub-modules in my project.
> 2. After working with it a while I then went back to a larger repository.
> 3. I deleted my sub-modules, but I did not clean up my `.gitmodules` file as specified [here](https://gist.github.com/myusuf3/7f645819ded92bda6677) or [here](https://davidwalsh.name/git-remove-submodule).  This left a lot of stray submodules in the `.gitmodules` file.
> **Note**: my `.git/config` file did not have any evidence of the sub-modules.
> 
> Once I cleaned up my `.gitmodules` file I was able to once again `Open Changes` and `Open File (HEAD)`.
> 
> ### How to test for the issue
> I had gone down a number of avenues, but a simple check was to delete the `.gitmodules` file and see if the issue went away.
